### PR TITLE
task/dynamic_layer/display_from_server

### PR DESCRIPTION
**Pull request by teamagam5**
Introduced and added the full flow of DynamicLayers

### CODE REVIEW

**Comment by teamagam5** [comment]
The tests encounter a memory problem (OOM exception) because of the 4GB RAM limit of circleci. The cause is the SQLite (room) tests which are performed in-memory and consume lots of it.

File: app/src/test/java/com/teamagam/gimelgimel/app/map/viewModel/DynamicLayersInteractorDisplayerTest.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    // Expect:
+    // createAdd(mEntity1) ->
+    // createRemove(mEntity1) ->
+    // createAdd(mEntity1) ->
+    // createAdd(mEntity2) .
+    verify(mMapSpy, times(2)).updateMapEntity(GeoEntityNotification.createAdd(mEntity1));
+    verify(mMapSpy).updateMapEntity(GeoEntityNotification.createRemove(mEntity1));
+    verify(mMapSpy).updateMapEntity(GeoEntityNotification.createAdd(mEntity2));
+  }
+}
```
Comment (BarAgam) [inline app/src/test/java/com/teamagam/gimelgimel/app/map/viewModel/DynamicLayersInteractorDisplayerTest.java:105]: <3

File: data/src/main/java/com/teamagam/gimelgimel/data/dynamicLayers/room/mapper/DynamicLayersEntityMapper.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  @Inject
+  public DynamicLayersEntityMapper(GeoFeatureEntityMapper geoFeatureEntityMapper) {
+    mGeoFeatureEntityMapper = geoFeatureEntityMapper;
+  }
+
+  public DynamicLayer mapToDomain(DynamicLayerEntity entity) {
+    if (entity == null) {
+      return null;
+    }
+    List<GeoEntity> list = extractEntities(entity);
```
Comment (BarAgam) [inline data/src/main/java/com/teamagam/gimelgimel/data/dynamicLayers/room/mapper/DynamicLayersEntityMapper.java]: entities?

File: data/src/main/java/com/teamagam/gimelgimel/data/dynamicLayers/room/mapper/DynamicLayersEntityMapper.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  public DynamicLayerEntity mapToEntity(DynamicLayer dynamicLayer) {
+    if (dynamicLayer == null) {
+      return null;
+    }
+
+    DynamicLayerEntity entity = new DynamicLayerEntity();
+
+    entity.id = dynamicLayer.getId();
+    entity.name = dynamicLayer.getName();
+    List<GeoEntity> domainEntities = dynamicLayer.getEntities();
```
Comment (BarAgam) [inline data/src/main/java/com/teamagam/gimelgimel/data/dynamicLayers/room/mapper/DynamicLayersEntityMapper.java]: extract code block to extract/convert/transform/mapToDataEntities

File: data/src/test/java/com/teamagam/gimelgimel/data/dynamicLayers/DynamicLayersDataRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+
+import static com.teamagam.gimelgimel.data.dynamicLayers.DynamicLayersTestUtils.assertEqualToStrings;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(RobolectricTestRunner.class)
+public class DynamicLayersDataRepositoryTest extends BaseTest {
```
Comment (BarAgam) [inline data/src/test/java/com/teamagam/gimelgimel/data/dynamicLayers/DynamicLayersDataRepositoryTest.java]: As we said, we'll have to comment this test out for now

**Review by BarAgam — APPROVED** [review]
Looks good.
We'll just have to comment out the robolectric tests for now unfortunately :<

**Pull request merged**